packages/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsProxy.cpp

```diff
@@ -331,9 +332,14 @@ void LayoutAnimationsProxy::handleUpdatesAndEnterings(
       }
 
       case ShadowViewMutation::Type::Update: {
-        auto shouldAnimate = hasLayoutChanged(mutation);
-        if (!layoutAnimationsManager_->hasLayoutAnimation(tag, LAYOUT) ||
-            (!shouldAnimate && !layoutAnimations_.contains(tag))) {
+        auto hasLayoutAnimation =
+            layoutAnimationsManager_->hasLayoutAnimation(tag, LAYOUT);
+        auto hasLayoutAndStyleAnimation =
+            layoutAnimationsManager_->hasLayoutAnimation(tag, LAYOUT_AND_STYLE);
+        auto layoutChanged = hasLayoutChanged(mutation);
+        if (!hasLayoutAnimation ||
+            (!hasLayoutAndStyleAnimation && !layoutChanged &&
+             !layoutAnimations_.contains(tag))) {
```

Comment:
Since this check is getting larger, could you move it to a separate method?

Reply:
I was trying several approaches and I've noticed, that the variables defined to get this checked (like `hasLayoutAndStyleAnimation`) are used later in the code. So extraction would bring some redundancy. Also I was wondering whether it doesn't make more sense to extract all long cases from this large switch loop (Update and Insert), and such a refactor could be safely done after merging this feature.